diff --git a/drivers/staging/media/imx/imx8-common.h b/drivers/staging/media/imx/imx8-common.h
index b7127047f9a4..179743d65365 100644
--- a/drivers/staging/media/imx/imx8-common.h
+++ b/drivers/staging/media/imx/imx8-common.h
@@ -39,6 +39,17 @@
 #define MXC_MIPI_CSI2_PAD_SINK			0
 #define MXC_MIPI_CSI2_PAD_SOURCE		1
 
+#define MXC_MIPI_CSI2_VC0_PAD_SINK		0
+#define MXC_MIPI_CSI2_VC1_PAD_SINK		1
+#define MXC_MIPI_CSI2_VC2_PAD_SINK		2
+#define MXC_MIPI_CSI2_VC3_PAD_SINK		3
+
+#define MXC_MIPI_CSI2_VC0_PAD_SOURCE		4
+#define MXC_MIPI_CSI2_VC1_PAD_SOURCE		5
+#define MXC_MIPI_CSI2_VC2_PAD_SOURCE		6
+#define MXC_MIPI_CSI2_VC3_PAD_SOURCE		7
+#define MXC_MIPI_CSI2_VCX_PADS_NUM		8
+
 /* Parallel CSI PADS */
 #define MXC_PARALLEL_CSI_PAD_SOURCE		0
 #define MXC_PARALLEL_CSI_PAD_SINK		1
